### PR TITLE
Fixes #23240 - BreadcrumbBar prop-type warnings

### PR DESCRIPTION
**Pull request by sharvit**

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/components/BreadcrumbSwitcherPopover.js [hunk 1]
```diff
@@ -60,10 +60,10 @@ const BreadcrumbSwitcherPopover = ({
...
         <ListGroup className="scrollable-list">
           {resources.map(resource => (
-              <ListGroupItem {...createItemProps(resource)}>
-                <EllipsisWithTooltip>{resource.name}</EllipsisWithTooltip>
-              </ListGroupItem>
-            ))}
+            <ListGroupItem {...createItemProps(resource)}>
+              <EllipsisWithTooltip>{resource.name}</EllipsisWithTooltip>
+            </ListGroupItem>
+          ))}
```
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/BreadcrumbBar/components/BreadcrumbSwitcherPopover.js:66]: Just prettier

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/components/BreadcrumbSwitcherPopover.js [hunk 2]
```diff
@@ -92,7 +92,7 @@ BreadcrumbSwitcherPopover.propTypes = {
   totalPages: PropTypes.number,
   onResourceClick: PropTypes.func,
   resources: PropTypes.arrayOf(PropTypes.shape({
-    id: PropTypes.string.isRequired,
+    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
```
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/BreadcrumbBar/components/BreadcrumbSwitcherPopover.js:95]: Actually fix the issue, `id` should be `string` or `number`

**Review by tbrisker — APPROVED** [review]
makes sense, thanks @sharvit !

**Comment by tbrisker** [comment]
cherry-picked to 1.18-stable as b6bc227c6

**Pull request merged**